kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -261,23 +262,47 @@ def foundry_kompile(
             shutil.copy(req, req_path)
             regen = True
 
-    if regen or not foundry_main_file.exists():
-        requires = ['foundry.md']
+    _imports: dict[str, list[str]] = {contract.name: [] for contract in foundry.contracts.values()}
+    for i in imports:
+        imp = i.split(':')
+        if not len(imp) == 2:
+            raise ValueError(f'module imports must be of the form "[ContractName]:[MODULE-NAME]". Got: {i}')
+        if imp[0] in _imports:
+            _imports[imp[0]].append(imp[1])
+        else:
+            raise ValueError(f'Could not find contract: {imp[0]}')
+
+    _LOGGER.info(f'foundry_contracts_file: {foundry_contracts_file}')
+    _LOGGER.info(f'foundry_main_file: {foundry_main_file}')
```

Comment:
In general, I follow the principle of:

- Log  that you are _about_ to read a given file (so that if it crashes at read time, you know what it was trying to do).
- Log that you _just did_ a file write (so that the log message only happens if the file got written).

Reply:
I removed these messages.